src/EnergyPlus/UtilityRoutines.cc

```diff
@@ -130,23 +127,26 @@ namespace UtilityRoutines {
         std::string::size_type const StringLen(PString.length());
         ErrorFlag = false;
         if (StringLen == 0) return rProcessNumber;
-        bool parseFailed = false;
         if (PString.find_first_not_of(ValidNumerics) == std::string::npos) {
             // make FORTRAN floating point number (containing 'd' or 'D')
             // standardized by replacing 'd' or 'D' with 'e'
             std::replace_if(
                 std::begin(PString), std::end(PString), [](const char c) { return c == 'D' || c == 'd'; }, 'e');
             // then parse as a normal floating point value
-            parseFailed = !readItem(PString, rProcessNumber);
-            ErrorFlag = false;
+            try {
+                rProcessNumber = std::stod(PString, nullptr);
+                ErrorFlag = false;
```

Comment:
ErrorFlag = false is already set at line 128. Delete line 138.  Also, remember the discussion long ago about setting a function argument bool &ErrorFlag to false within any function. It's OK to set it true, but don't set it to false which would erase any previous error. To that end I would delete line 128 as well. @mjwitte.

I am amazed at the impact this has on execution speed!

Reply:
It looks like there are several places in code where this function is called that relies on ErrorFlag being true or false for each specific call. So do not delete line 128,